tests/Utilities/assertingUtils.py

```diff
@@ -0,0 +1,174 @@
+    LOG.info("Parsing %r with %r", optionName, value)
+    if isinstance(value, basestring) and value.lower() in ('no', 'false'):
+      outDict[optionName] = False
+    elif isinstance(value, basestring) and value.lower() in ('yes', 'true'):
+      outDict[optionName] = True
+    elif isinstance(value, basestring) and ',' in value:
+      outDict[optionName] = [val.strip() for val in value.split(',')]
+    elif isinstance(value, dict):
+      _parseOption(outDict, value, optionPrefix=optionName)
+    elif isinstance(value, basestring):
+      outDict[optionName] = value
+      if value.isdigit():
+        try:
+          outDict[optionName] = int(value)
+        except ValueError:
+          pass
+      else:
+        try:
+          outDict[optionName] = float(value)
+        except ValueError:
+          pass
+
+
+def AgentOptionsTest(agentPath, ignoreOptions, mocker):
+  """Test the consistency of options in ConfigTemplate and initialize of the agent.
+
+  :param str agentPath: Module where the agent can be found, e.g. DIRAC.Core.Agent.CoreAgent
+  :param list ignoreOptions: list of options to ignore during checks
+  :param mocker: the mocker fixture from pytest-mock
+  """
+  agentPathSplit = agentPath.split('.')
+  extension = agentPathSplit[0]  # DIRAC or VODIRAC, etc.
+  systemName = agentPathSplit[1]
+  agentName = agentPathSplit[-1]
+
+  agentModule = importlib.import_module(agentPath)
+  LOG.info("Agents: %s %s", agentPath, agentModule)
+  agentClass = None
+
+  # mock everything but the agentClass
+  for name, member in inspect.getmembers(agentModule):
+    LOG.info("Mocking? %s, %s, %s, isclass(%s)", name, callable(member), type(member), inspect.isclass(member))
+    if name != 'AgentModule' and '_AgentModule__executeModuleCycle' in dir(member):
+      LOG.info("Found the agent class %s, %s", name, member)
+      agentClass = member
+      continue
+    elif name == 'AgentModule':
+      continue
+    if callable(member) or inspect.ismodule(member):
+      LOG.info("Mocking: %s, %s, %s", name, member, type(member))
+      mocker.patch(agentPath + "." + name, new=Mock())
+
+  if hasattr(agentModule, 'gConfig'):
+    gConfigMock = Mock()
+    gConfigMock.getSections.return_value = dict(OK=True, Value=[])
+    mocker.patch(agentPath + "." + 'gConfig', new=gConfigMock)
+
+  def returnDefault(*args):
+    LOG.debug("ReturningDefault: %s, %s", args, type(args[1]))
+    return args[1]
+
+  getOptionMock = Mock(name="am_getOption", side_effect=returnDefault)
+
+  def instrument(*args, **kwargs):
+    """Mock some functions that come from the AgentModule and are not present otherwise."""
+    args[0].am_getControlDirectory = Mock()
+    args[0].am_getOption = getOptionMock
+    args[0].am_getModuleParam = Mock()
+    args[0].am_getWorkDirectory = Mock()
+    args[0].am_setOption = Mock()
+    args[0].log = Mock()
+    return None
+  initMock = Mock(side_effect=instrument)
+
+  class MockAgentModule(object):
```

Comment:
Is this used?

Reply:
Yes, some agents have an `__init__` function called where the AgentModule is called with `super`, then we need to inject the base class with other one, other call AgentModule.__init__ directly (which we mock as well), third ones don't have any __init__, which needs another way of mocking things

Reply:
Ah I see. I think this is technically not a mock but a "fake", but it doesn't really matter.